pallets/collator-selection/src/lib.rs

```diff
@@ -377,11 +377,11 @@ pub mod pallet {
 		pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T>
 	{
 		fn note_author(author: T::AccountId) {
-			let treasury = Self::account_id();
-			let reward = T::Currency::free_balance(&treasury).div(2u32.into());
-
-			// `reward` is half of treasury account, this should never fail.
-			let _success = T::Currency::transfer(&treasury, &author, reward, KeepAlive);
+			let pot = Self::account_id();
+			// assumes an ED will be sent to pot.
+			let reward = T::Currency::free_balance(&pot).checked_sub(&T::Currency::minimum_balance()).unwrap_or_else(Zero::zero).div(2u32.into());
+			// `reward` is half of pot account, this should never fail.
+			let _success = T::Currency::transfer(&pot, &author, reward, KeepAlive);
```

Comment:
Should we either add a test that this is a `noop` if `reward = 0`, or put `if reward > 0 { transfer }`?